gammapy/maps/axes.py

```diff
@@ -2545,15 +2545,16 @@ def pix_to_coord(self, pix):
             Array of axis coordinate values.
         """
         shape = np.shape(pix)
-        pix = np.atleast_1d(pix).ravel()
-        coords = np.zeros(len(pix))
+        pix = np.atleast_1d(pix)
+        coords = np.zeros_like(pix)
         frac, idx = np.modf(pix)
+        idx1 = idx.astype(int)
         valid = np.logical_and(idx >= 0, idx < self.nbin, np.isfinite(idx))
         idx_valid = np.where(valid)
         idx_invalid = np.where(~valid)
 
         coords[idx_valid] = (
-            frac[idx_valid] * self.time_delta[idx_valid] + self.edges_min[idx_valid]
+            frac[idx_valid] * self.time_delta[idx1[valid]] + self.edges_min[idx1[valid]]
         ).jd
```

Comment:
I don't completely understand why converting to `.jd` and then multiplying by `self.units` works. Aren't there problems when units are non-standard?

Reply:
What do you mean "non-standard units"? TimeAxis must have units of time, no?

Reply:
Thanks @cgalelli ! I changed it to use .value and then multiplying by self.units should be consistent

Reply:
Perfect! Sorry if I was not clear in the first message, but this solution was exactly what I meant.